src/librustc/traits/select.rs

```diff
@@ -2056,8 +2044,8 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 Ok(VtableFnPointer(data))
             }
 
-            ProjectionCandidate => {
-                self.confirm_projection_candidate(obligation);
+            ProjectionCandidate(matching_bound) => {
```

Comment:
May I suggest replacing `matching_bound` everywhere you used it with just `bound`? It keeps the code shorter.